### ISSUE TITLE
Error In This Lesson Basic JavaScript: Escape Sequences in Strings

### ISSUE THREAD

**Issue by AxiomInfinite** [freeCodeCamp/freeCodeCamp#17712]
#### Describe your problem and - if possible - how to reproduce it

Lesson Completes with this code
Error Code thats being accepted:

var myStr = 'FirstLine\n\tSecondLine\nThirdLine'; 

Correct Code:

var myStr = 'FirstLine\n\t\\SecondLine\nThirdLine'; 

#### Add a Link to the page with the problem

https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/escape-sequences-in-strings
#### Tell us about your browser and operating system

* Browser Name: Chrome 
* Browser Version: 
* Operating System:

**Comment by scissorsneedfoodtoo** [comment]
@rbonner19, thank you for submitting this issue! Great catch.

You're right, the test for this challenge is wrong, so incorrect code still passes. The correct code should be:

`var myStr = 'FirstLine\n\t\\SecondLine\nThirdLine';`

Would you be interested in fixing this? You could do so over in the new Curriculum repo the challenge seed files were recently moved to. Here's the line that needs to be changed:

https://github.com/freeCodeCamp/curriculum/blob/34f45d25fb0f2b6959539a892892e48b5768fc57/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json#L1551

I believe the line should be changed to:

`"assert(/\\\\SecondLine/.test(myStr), '<code>SecondLine</code> should be preceded by the backslash character <code>\\\\</code>');"`

Would you be interested in submitting a pull request to fix this test?

**Comment by zsoltime** [comment]
If @rbonner19 not interested in submitting a PR, I'm happy to fix this. 😉

**Comment by rbonner19** [comment]
Yes I'd love to submit the fix, should be able to do that tonight!

**Comment by scissorsneedfoodtoo** [comment]
Okay @rbonner19, that's great! Looking forward to your PR later.

**Comment by Nirajn2311** [comment]
@AxiomInfinite are you working on the fix?

**Comment by randelldawson** [comment]
I could be wrong, but it seems the following test:
````
        {
          "text":
            "<code>SecondLine</code> should be preceded by the backslash character <code>\\\\</code>",
          "testString":
            "assert(/\\SecondLine/.test(myStr), '<code>SecondLine</code> should be preceded by the backslash character <code>\\\\</code>');"
        },
```
needs to be changed to:
```
        {
          "text":
            "<code>SecondLine</code> should be preceded by the backslash character <code>\\\\</code>",
          "testString":
            "assert(/\\\\SecondLine/.test(myStr), '<code>SecondLine</code> should be preceded by the backslash character <code>\\\\</code>');"
        },
```

**Comment by RandellDawson** [comment]
@GT1man - Your code is incorrect.  These types of issues you are having are better discussed on the forum.  You have a typo.  JavaScript is case-sensitive.  If you are still having problems with this hint, click the Ask for Help button on the challenge.

**Comment by RandellDawson** [comment]
Finally got around to creating a PR (#35367) to prevent incorrect solutions, such as the one @AxiomInfinite, from being accepted as a valid solution to the challenge.

**Comment by Manish-Giri** [comment]
I tested this locally, and the last test pretty much ensures there's only one (correct) way to pass the challenge.

I did notice that with this string - `'FirstLine\n\t\\SecondLine\n\n\ThirdLine';`, this test doesn't pass - 
> There should be a newline character between SecondLine and ThirdLine

Based on the test requirement, there _is_, technically, a newline between `SecondLine` and `ThirdLine`. So, one could ask why the test won't pass. Maybe the text assert could be changed to - 

> There should be **one** newline character between SecondLine and ThirdLine

But that's probably beyond the scope of this PR, and a good candidate for another PR.

Either way, LGTM.
